RecoHI/HiEvtPlaneAlgos/src/EvtPlaneProducer.cc

```diff
@@ -227,44 +368,90 @@ EvtPlaneProducer::EvtPlaneProducer(const edm::ParameterSet &iConfig)
       caloTag_(iConfig.getParameter<edm::InputTag>("caloTag")),
       castorTag_(iConfig.getParameter<edm::InputTag>("castorTag")),
       trackTag_(iConfig.getParameter<edm::InputTag>("trackTag")),
+      losttrackTag_(iConfig.getParameter<edm::InputTag>("lostTag")),
+      chi2MapTag_(iConfig.getParameter<edm::InputTag>("chi2MapTag")),
+      chi2MapLostTag_(iConfig.getParameter<edm::InputTag>("chi2MapLostTag")),
       loadDB_(iConfig.getParameter<bool>("loadDB")),
       minet_(iConfig.getParameter<double>("minet")),
       maxet_(iConfig.getParameter<double>("maxet")),
       minpt_(iConfig.getParameter<double>("minpt")),
       maxpt_(iConfig.getParameter<double>("maxpt")),
       minvtx_(iConfig.getParameter<double>("minvtx")),
       maxvtx_(iConfig.getParameter<double>("maxvtx")),
-      dzerr_(iConfig.getParameter<double>("dzerr")),
+      flatnvtxbins_(iConfig.getParameter<int>("flatnvtxbins")),
+      flatminvtx_(iConfig.getParameter<double>("flatminvtx")),
+      flatdelvtx_(iConfig.getParameter<double>("flatdelvtx")),
+      dzdzerror_(iConfig.getParameter<double>("dzdzerror")),
+      d0d0error_(iConfig.getParameter<double>("d0d0error")),
+      pterror_(iConfig.getParameter<double>("pterror")),
+      chi2perlayer_(iConfig.getParameter<double>("chi2perlayer")),
+      dzdzerror_pix_(iConfig.getParameter<double>("dzdzerror_pix")),
       chi2_(iConfig.getParameter<double>("chi2")),
+      nhitsValid_(iConfig.getParameter<int>("nhitsValid")),
       FlatOrder_(iConfig.getParameter<int>("FlatOrder")),
       NumFlatBins_(iConfig.getParameter<int>("NumFlatBins")),
       caloCentRef_(iConfig.getParameter<double>("caloCentRef")),
       caloCentRefWidth_(iConfig.getParameter<double>("caloCentRefWidth")),
-      CentBinCompression_(iConfig.getParameter<int>("CentBinCompression")) {
+      CentBinCompression_(iConfig.getParameter<int>("CentBinCompression")),
+      cutEra_(iConfig.getParameter<int>("cutEra"))
+
+{
+  switch (cutEra_) {
+    case 0:
+      cuts = new EPCuts(
+          EP_ERA::ppReco, pterror_, dzdzerror_, d0d0error_, chi2perlayer_, dzdzerror_pix_, chi2_, nhitsValid_);
+      break;
+    case 1:
+      cuts = new EPCuts(
+          EP_ERA::HIReco, pterror_, dzdzerror_, d0d0error_, chi2perlayer_, dzdzerror_pix_, chi2_, nhitsValid_);
+      break;
+    case 2:
+      cuts = new EPCuts(
+          EP_ERA::Pixel, pterror_, dzdzerror_, d0d0error_, chi2perlayer_, dzdzerror_pix_, chi2_, nhitsValid_);
+      break;
+    case 3:
+      cuts = new EPCuts(
+          EP_ERA::GenMC, pterror_, dzdzerror_, d0d0error_, chi2perlayer_, dzdzerror_pix_, chi2_, nhitsValid_);
+      break;
+    default:
+      cuts = nullptr;
+  }
```

Comment:
```suggestion
  if (cutEra_ > 3)
      throw edm::Exception(edm::errors::Configuration)<<"wrong range in cutEra parameter";
  cuts_ = EPCuts(
          static_cast<EP_ERA>(cutEra_), pterror_, dzdzerror_, d0d0error_, chi2perlayer_, dzdzerror_pix_, chi2_, nhitsValid_);
  }
```
there is no need for `cuts` to be a pointer